File: packages/dart_core/lib/models/enums.dart
enum UserRole {
  superAdmin('SUPER_ADMIN'),
  principal('PRINCIPAL'),
  faculty('FACULTY'),
  student('STUDENT');

  final String value;
  const UserRole(this.value);

  static UserRole fromString(String val) {
    return UserRole.values.firstWhere(
      (e) => e.value == val,
      orElse: () => UserRole.student,
    );
  }
}

enum FacultyStatus {
  available('AVAILABLE'),
  busy('BUSY'),
  inClass('IN_CLASS'),
  meeting('MEETING'),
  onBreak('ON_BREAK'),
  dnd('DND'),
  offline('OFFLINE');

  final String value;
  const FacultyStatus(this.value);

  static FacultyStatus fromString(String val) {
    return FacultyStatus.values.firstWhere(
      (e) => e.value == val,
      orElse: () => FacultyStatus.offline,
    );
  }
}

enum PresenceSource {
  dynamicQr('DYNAMIC_QR'),
  teacherScan('TEACHER_SCAN'),
  studentQr('STUDENT_QR'),
  timetable('TIMETABLE'),
  p2pProximity('P2P_PROXIMITY'),
  sensor('SENSOR'),
  manualOverride('MANUAL_OVERRIDE');

  final String value;
  const PresenceSource(this.value);

  static PresenceSource fromString(String val) {
    return PresenceSource.values.firstWhere(
      (e) => e.value == val,
      orElse: () => PresenceSource.dynamicQr,
    );
  }
}

enum SyncStatus {
  pending('PENDING'),
  syncing('SYNCING'),
  synced('SYNCED'),
  failed('FAILED'),
  rejected('REJECTED');

  final String value;
  const SyncStatus(this.value);
}

File: backend/src/constants/enums.ts
export enum UserRole {
  SUPER_ADMIN = 'SUPER_ADMIN',
  PRINCIPAL = 'PRINCIPAL',
  FACULTY = 'FACULTY',
  STUDENT = 'STUDENT'
}

export enum FacultyStatus {
  IN_CLASS = 'IN_CLASS', // Verified physical scan in room
  NOT_IN_CLASS_PENDING_SCAN = 'NOT_IN_CLASS_PENDING_SCAN', // Scheduled in timetable but not yet scanned room QR
  WITH_PRINCIPAL = 'WITH_PRINCIPAL', // Detected / in meeting with Principal
  AVAILABLE = 'AVAILABLE', // Available in Staff Room / Cabin
  BUSY = 'BUSY',
  MEETING = 'MEETING',
  ON_BREAK = 'ON_BREAK',
  MOBILE_DATA_OFF_BLE_ONLY = 'MOBILE_DATA_OFF_BLE_ONLY',
  DND = 'DND',
  OFFLINE = 'OFFLINE'
}

export enum RoomStatus {
  AVAILABLE = 'AVAILABLE',
  OCCUPIED = 'OCCUPIED',
  MAINTENANCE = 'MAINTENANCE'
}

export enum ClassSessionStatus {
  SCHEDULED = 'SCHEDULED',
  ACTIVE = 'ACTIVE',
  PAUSED = 'PAUSED',
  ENDED = 'ENDED',
  CANCELLED = 'CANCELLED'
}

export enum AttendanceStatus {
  PRESENT = 'PRESENT',
  UNCERTAIN = 'UNCERTAIN',
  ABSENT = 'ABSENT',
  LEAVE = 'LEAVE',
  EXCUSED = 'EXCUSED'
}

export enum PresenceState {
  HIGH_CONFIDENCE = 'HIGH_CONFIDENCE',
  PROBABLE = 'PROBABLE',
  UNCERTAIN = 'UNCERTAIN',
  NOT_DETECTED = 'NOT_DETECTED',
  UNKNOWN = 'UNKNOWN'
}

export enum PresenceSource {
  DYNAMIC_QR = 'DYNAMIC_QR',
  TEACHER_SCAN = 'TEACHER_SCAN',
  STUDENT_QR = 'STUDENT_QR',
  TIMETABLE = 'TIMETABLE',
  P2P_PROXIMITY = 'P2P_PROXIMITY',
  BLE_BEACON = 'BLE_BEACON',
  SENSOR = 'SENSOR',
  MANUAL_OVERRIDE = 'MANUAL_OVERRIDE'
}

export enum SyncEventStatus {
  PENDING = 'PENDING',
  SYNCING = 'SYNCING',
  PROCESSED = 'PROCESSED',
  DUPLICATE = 'DUPLICATE',
  INVALID_SIGNATURE = 'INVALID_SIGNATURE',
  REJECTED = 'REJECTED',
  FAILED = 'FAILED'
}

export enum MeetingRequestStatus {
  PENDING = 'PENDING',
  ACCEPTED = 'ACCEPTED',
  DECLINED = 'DECLINED',
  RESCHEDULED = 'RESCHEDULED',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED'
}

export enum LeaveRequestStatus {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED',
  CANCELLED = 'CANCELLED'
}

export enum DevicePlatform {
  ANDROID = 'ANDROID',
  IOS = 'IOS',
  WEB = 'WEB',
  DESKTOP = 'DESKTOP'
}

export enum MovementState {
  STATIONARY = 'STATIONARY',
  WALKING = 'WALKING',
  MOVING = 'MOVING',
  UNKNOWN = 'UNKNOWN'
}

export enum SocketEvent {
  CLASS_STARTED = 'class.started',
  CLASS_ENDED = 'class.ended',
  ATTENDANCE_UPDATED = 'attendance.updated',
  PRESENCE_UPDATED = 'presence.updated',
  FACULTY_STATUS_CHANGED = 'faculty.status.changed',
  MEETING_CREATED = 'meeting.created',
  MEETING_UPDATED = 'meeting.updated',
  ROOM_CHANGED = 'room.changed',
  TIMETABLE_UPDATED = 'timetable.updated',
  NOTIFICATION_CREATED = 'notification.created',
  SYNC_COMPLETED = 'sync.completed',
  BLE_ALERT_DISPATCHED = 'ble.alert.dispatched'
}
